### PR TITLE
Fix: Index out of bounds for incomplete attribute value

### PR DESCRIPTION
**Pull request by ajaybhargavb**
#555 
@NTaylorMullen

### CODE REVIEW

File: src/Microsoft.AspNet.Razor/Parser/TagHelpers/TagHelperBlockRewriter.cs [hunk 1]
```diff
@@ -215,34 +215,28 @@ public static class TagHelperBlockRewriter
...
+                        symbolStartLocation = htmlSymbols[i].Start;
 
                         // If there's a start quote then there must be an end quote to be valid, skip it.
                         symbolOffset = 1;
-
-                        i = valueStartIndex - 1;
                     }
                     else
                     {
+                        // No quote found. Go back.
```
Comment (NTaylorMullen) [inline src/Microsoft.AspNet.Razor/Parser/TagHelpers/TagHelperBlockRewriter.cs]: Clarify comment. This `i--` ensures we don't skip past the value after the `=`. Such as: `<input checked=true>` would result in us skipping the `true` symbol if we did not have this `i--`.

**Comment by NTaylorMullen** [comment]
:shipit:

**Comment by Eilon** [comment]
Please send mail ASAP for approval to `release` (and then rebase this on `release`, then merge back to `dev`).

**Comment by ajaybhargavb** [comment]
bc103fcaa404c3fabd902259e170f18f2f183738

**Pull request closed without merging**